app/scripts/modules/core/src/managed/ManagedResourceHistoryModal.tsx

```diff
@@ -0,0 +1,227 @@
+  resourceSummary: IManagedResourceSummary;
+}
+
+const { useMemo } = React;
+
+const viewConfigurationByEventType = {
+  ResourceCreated: {
+    displayName: 'Created',
+    iconClass: 'icon-md-created',
+    level: 'info',
+  },
+  ResourceUpdated: {
+    displayName: 'Config updated',
+    // Needs it's own icon
+    iconClass: 'icon-md-created',
+    level: 'info',
+  },
+  ResourceDeleted: {
+    displayName: 'Deleted',
+    // Needs it's own icon
+    iconClass: 'icon-md-missing-resource',
+    level: 'info',
+  },
+  ResourceMissing: {
+    displayName: 'Missing',
+    iconClass: 'icon-md-missing-resource',
+    level: 'info',
+  },
+  ResourceValid: {
+    displayName: 'Valid',
+    // Needs it's own icon
+    iconClass: 'icon-md-delta-resolved',
+    level: 'info',
+  },
+  ResourceDeltaDetected: {
+    displayName: 'Drift detected',
+    iconClass: 'icon-md-delta-detected',
+    level: 'info',
+  },
+  ResourceDeltaResolved: {
+    displayName: 'Drift resolved',
+    iconClass: 'icon-md-delta-resolved',
+    level: 'info',
+  },
+  ResourceActuationLaunched: {
+    displayName: 'Task launched',
+    iconClass: 'icon-md-actuation-launched',
+    level: 'info',
+  },
+  ResourceCheckError: {
+    displayName: 'Error',
+    // Needs it's own icon
+    iconClass: 'icon-md-error',
+    level: 'error',
+  },
+  ResourceCheckUnresolvable: {
+    displayName: 'Error',
+    // Needs it's own icon, but could likely be same as ResourceCheckError
+    iconClass: 'icon-md-error',
+    level: 'error',
+  },
+  ResourceActuationPaused: {
+    displayName: 'Management paused',
+    // Needs it's own icon
+    iconClass: 'icon-md-paused',
+    level: 'warning',
+  },
+  ResourceActuationResumed: {
+    displayName: 'Management resumed',
+    // Needs it's own icon
+    iconClass: 'icon-md-resumed',
+    level: 'info',
+  },
+} as const;
+
+const renderExpandedRowContent = (
+  level: 'info' | 'warning' | 'error',
+  diff: IManagedResourceDiff,
+  tasks: Array<{ id: string; name: string }>,
+  message: string,
+) => {
+  return (
+    <div className="flex-container-v left">
+      {message && (
+        <div
+          className={classNames('sp-padding-xs-yaxis', {
+            'event-warning-message': level === 'warning',
+            'event-error-message': level === 'error',
+          })}
+        >
+          {message}
+        </div>
+      )}
+      {tasks && (
+        <div className="flex-container-v">
+          {tasks.map(({ id, name }) => (
+            <UISref key={id} to="home.applications.application.tasks.taskDetails" params={{ taskId: id }}>
+              <a className="sp-padding-xs-yaxis">{name}</a>
+            </UISref>
+          ))}
+        </div>
+      )}
+      {diff && <ManagedResourceDiffTable diff={diff} />}
+    </div>
+  );
+};
+
+export const ManagedResourceHistoryModal = ({ resourceSummary }: IManagedResourceHistoryModalProps) => {
+  const {
+    id,
+    moniker: { app, stack, detail },
```

Comment:
Since you're joining the moniker parts into a single name just below, and considering that @robfletcher is in the process of removing monikers, would it make sense to parse the API result directly into a single name property?

Reply:
We generally prefer using `moniker` objects as the canonical representation in UI code — 9 times out of 10 the non-trivial things we want to do with names/monikers are super inconvenient if you have to constantly parse out the name into an object (and then give it a name and pass it around, etc. etc.). Joining on `-` for actually plopping the name on the screen at the end of all the complicated data massaging is the easy part 😉